salt/modules/status.py

```diff
@@ -366,8 +366,12 @@ def nproc():
 
         salt '*' status.nproc
     '''
-    data = __salt__['cmd.run']('nproc')
+    cmd = {
+        'Linux': 'nproc',
+        'FreeBSD': 'sysctl -n hw.ncpu',
+    }
     try:
+        data = __salt__['cmd.run'](cmd[__grains__['kernel']])
```

Comment:
Right now the approach is good, we would just add them into the cmd dict you have made